### PR TITLE
feat: add icon decisions

### PR DESCRIPTION
**Pull request by chaitanyadeorukhkar**
### [View Formatted Decision](https://github.com/razorpay/blade/blob/feat/add-icon-decisions/packages/blade/src/components/Icons/_decisions/decisions.md)

### CODE REVIEW

**Review by divyanshu013 — COMMENTED** [review]
Will we provide an API for consumers to create custom icons? Products may have some unique icons, so a consistent API would be nice to have.

**Comment by chaitanyadeorukhkar** [comment]
> Will we provide an API for consumers to create custom icons? Products may have some unique icons, so a consistent API would be nice to have.

Products could directly use SVG for such edge cases where they need custom icons. We're creating this wrapper just to solve the problem of writing icons twice on blade, otherwise the native svg on web & `react-native-svg` on native would be good enough to help create their custom icons

Maybe later when we see a lot of custom icons being created, we can figure out if we can add those custom icons to blade or make export the internal `_SVG` component at that point

**Comment by chaitanyadeorukhkar** [comment]
> Works. For inspiration [`createIcon`](https://chakra-ui.com/docs/components/media-and-icons/icon#using-the-createicon-function) utility from `chakra-ui`.

I might be biased but I find the `createIcon` way of doing things more complicated than our way. In our case, our devs would need to simply 
1. copy the svg from figma
2. paste it in svgr (we can skip this by adding a svgr plugin to figma so that its copied in transformed format)
3. Change imports to internal `_SVG`

With the `createIcon` process, our devs would need to pass a lot of information down in a specific format.

**Comment by chaitanyadeorukhkar** [comment]
@divyanshu013 chakra also allows you to create components with their `Icon` component. [See this](https://chakra-ui.com/docs/components/media-and-icons/icon#using-the-icon-component). Its exactly the same way we're doing with our Svg component, we're just calling it what it is - SVG. They have a note that says "The Icon component renders as an svg element." 😅

**Review by divyanshu013 — APPROVED** [review]
Looks good to me.

File: packages/blade/src/components/Icons/_decisions/decisions.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  >
+  ```
+  
+- Each icon will have the same `height` & `width` which would be derived from its `size` prop. Here's what each size corresponds to
+  - xxsmall: `width='10px'`, `height='10px'`
+  - xsmall: `width='12px'`, `height='12px'`
+  - small: `width='16px'`, `height='16px'`
+  - medium: `width='20px'`, `height='20px'`
+  - large: `width='24px'`, `height='24px'`
+  - xlarge: `width='30px'`, `height='30px'`
```
[the remark this answers is not in the document]
Comment (chaitanyadeorukhkar) [reply packages/blade/src/components/Icons/_decisions/decisions.md:44]: Done

**Review by kamleshchandnani — APPROVED** [review]

**Pull request merged**